src/LayerEditor/ui/data/block_model.py

```diff
@@ -0,0 +1,63 @@
+from LayerEditor.data.geometry_model import LayerGeometryModel
+from LayerEditor.ui.data.layer_model import LayerModel
+
+from LayerEditor.ui.diagram_editor.diagram_scene import DiagramScene
+from LayerEditor.ui.tools.selection import Selection
+
+import gm_base.polygons.polygons_io as polygons_io
+from gm_base.geometry_files.format_last import InterfaceNodeSet
+
+
+class BlockModel:
+    """Holds common data for a block of layers"""
+    def __init__(self, top_idx, le_data):
+        self.le_data = le_data
+        """Reference for LEData."""
+        self.top_idx = top_idx
```

Comment:
The topology and nodeset indices are useless during operation and only replace references in serialization.